src/main/webapp/app/entities/quiz-exercise/quiz-exercise-detail.directive.ts

```diff
@@ -326,6 +435,81 @@ class QuizExerciseDetailController {
         return reasonString.substr(0, reasonString.length - 5);
     }
 
+    /**
+     * Imports a json quiz file and adds questions to current quiz exercise.
+     */
+    async importQuiz() {
+        if (this.importFile === null || this.importFile === undefined) {
+            return;
+        }
+        const fileReader = new FileReader();
+        fileReader.onload = () => {
+            try {
+                const questions = JSON.parse(fileReader.result);
+                this.addQuestions(questions);
+                this.importFile = null;
+                const control = document.getElementById('importFileInput') as HTMLInputElement;
+                control.value = null;
+            } catch (e) {
+                alert('Import Quiz Failed! Invalid quiz file.');
+            }
+        };
+        fileReader.readAsText(this.importFile);
+    }
+
+    /**
+     * Adds given questions to current quiz exercise.
+     * Ids are removed from new questions so that new id is assigned upon saving the quiz exercise.
+     * Images are duplicated for drag and drop questions.
+     * @param questions list of questions
+     */
+    async addQuestions(questions: any) {
```

Comment:
please add more comments in the function itself.
Why is the param any? Please avoid any because it does not guarantee any type-safety and rather use a proper type here!

Reply:
Have a look at this commit. I added tempID in required models,
https://github.com/ls1intum/ArTEMiS/pull/85/commits/4bf784d2bc00f24443cadf6e0c353875d7b4ef92